src/psyclone/tests/psyir/transformations/transformations_test.py

```diff
@@ -276,6 +277,49 @@ def test_omptargettrans(sample_psyir):
     assert loops[0].parent.parent is loops[1].parent.parent
 
 
+def test_omptargettrans_validate(sample_psyir):
+    ''' Test that OMPTargetTrans validation fails if it contains non-allowed
+    constructs. '''
+
+    omptargettrans = OMPTargetTrans()
+    tree = sample_psyir.copy()
+    loops = tree.walk(Loop, stop_type=Loop)
+
+    # Valid loop
+    omptargettrans.validate(loops[0])
+
+    # With a CodeBlock it should fail
+    loops[0].loop_body.addchild(CodeBlock([], CodeBlock.Structure.STATEMENT))
+    with pytest.raises(TransformationError) as err:
+        omptargettrans.validate(loops[0])
+    assert ("Nodes of type 'CodeBlock' cannot be enclosed by a OMPTargetTrans "
+            "transformation" in str(err.value))
+
+    # With a LBOUND it should fail
+    symbol = DataSymbol("dummy", INTEGER_TYPE)
+    loops[0].loop_body[-1].replace_with(
+            Assignment.create(
+                Reference(symbol),
+                BinaryOperation(BinaryOperation.Operator.LBOUND)))
+    with pytest.raises(TransformationError) as err:
+        omptargettrans.validate(loops[0])
+    assert ("The OMPTargetDirective can not be inserted as an ancestor of "
```

Comment:
Please could you change this and the following error message to the more user-friendly form used for 'CodeBlock' - i.e. talk about 'containing' rather than 'ancestors'.

Reply:
error no longer exists